sovrin_client/agent/agent.py

```diff
@@ -220,14 +220,21 @@ def createAgent(agentClass, name, wallet=None, basedirpath=None, port=None,
                          ha=("0.0.0.0", clientPort),
                          basedirpath=basedirpath)
 
-    return agentClass(basedirpath=basedirpath,
-                      client=client,
-                      wallet=wallet,
-                      port=port,
-                      loop=loop)
+    try:
+        return agentClass(basedirpath=basedirpath,
+                          client=client,
+                          wallet=wallet,
+                          port=port,
+                          loop=loop)
+    except AddressAlreadyInUse as e:
+        msg = getFormattedErrorMsg(str(e))
+        logger.error(msg)
 
 
 def runAgent(agent, looper=None, bootstrap=True):
+    if not agent:
+        return
+
```

Comment:
This will be confusing to a caller because the value returned will be None. Let's leverage Exceptions where appropriate. A line that reads like this would be great:
`    assert agent`